src/main/java/com/google/api/codegen/transformer/SampleTransformer.java

```diff
@@ -44,18 +47,38 @@
  * A class that performs the transformations needed to generate the MethodSampleView for the
  * applicable sample types.
  */
-public class SampleTransformer {
+@AutoValue
+public abstract class SampleTransformer {
```

Comment:
It is a little bit odd to me that we have an autovalue class that also contains a lot of complicated logic. I'm not sure whether we do this in other places in gapic-generator. But I don't have any concrete suggestions for change... :-/

Reply:
How about handwriting the fields and a builder?